src/lsst/cmservice/handlers/script_handler.py

```diff
@@ -384,6 +359,42 @@ async def _check_slurm_job(  # pylint: disable=unused-argument
             await script.update_values(session, status=status)
         return status
 
+    async def _check_htcondor_job(  # pylint: disable=unused-argument
+        self,
+        session: async_scoped_session,
+        htcondor_id: str | None,
+        script: Script,
+        parent: ElementMixin,
+    ) -> StatusEnum:
+        """Check the status of a `Script` sent to htcondor
+
+        Parameters
+        ----------
+        session : async_scoped_session
+            DB session manager
+
+        htcondor_id : str
+            HTCondor job id
+
+        script: Script
+            The `Script` in question
+
+        parent: ElementMixin
+            Parent Element of the `Script` in question
+
+        Returns
+        -------
+        status : StatusEnum
+            The status of the processing
+        """
+        status = await check_htcondor_job(htcondor_id)
+        print(f"Getting status for {script.fullname} {status}")
+        if status is None:
+            status = StatusEnum.running
```

Comment:
Is it worth coding an upper limit for how long status is allowed to be None, since getting status None seems to imply that we don't find the job in HTCondor?

Reply:
Nice catch, actually, the docstring there is a bit wrong, as it was copied from slurm.   check_htcondor_job should never actually return None.  I will fix this.